api/src/main/kotlin/io/spine/protodata/renderer/Renderer.kt

```diff
@@ -74,4 +75,28 @@ protected constructor(
     final override fun <T> configAs(cls: Class<T>): T = super.configAs(cls)
 
     final override fun configIsPresent(): Boolean = super.configIsPresent()
+
+    /**
+     * Injects the context of the ProtoData application.
+     *
+     * This method is `public` but is essentially `internal` to ProtoData SDK.
+     *
+     * @see 
+     */
+    public override fun registerWith(protoDataContext: BoundedContext) {
+        if (isRegistered) {
+            check(this.protoDataContext == protoDataContext) {
+                "Unable to register the renderer `$this` with" +
+                        " `${protoDataContext.name().value}`." +
+                        " The renderer is already registered with" +
+                        " `${this.protoDataContext.name().value}`."
+            }
+            return
+        }
```

Comment:
Please use `checkNotRegistered()` instead.

Reply:
I thought of this, but `checkNotRegistered()` does not give enough diagnostics. It simply tells that an object is already registered. We need to tell a bit more in such a delicate case.